travis test failure

https://travis-ci.org/coreos/etcd/builds/43783487

panic: test timed out after 1m0s
goroutine 5493 [running]:
runtime.panic(0x862f60, 0xc2085823f0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/panic.c:279 +0xf5
testing.func·008()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:629 +0x12e
created by time.goFunc
    /home/travis/.gvm/gos/go1.3.3/src/pkg/time/sleep.go:121 +0x55
goroutine 16 [chan receive]:
testing.RunTests(0xa93c68, 0xc5d7e0, 0x19, 0x19, 0x1)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:505 +0xbb8
testing.Main(0xa93c68, 0xc5d7e0, 0x19, 0x19, 0xc79180, 0x0, 0x0, 0xc79180, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:435 +0xa3
main.main()
    github.com/coreos/etcd/integration/_test/_testmain.go:141 +0x18c
goroutine 19 [finalizer wait]:
runtime.park(0x4141e0, 0xc738d8, 0xc5fe49)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/proc.c:1369 +0x89
runtime.parkunlock(0xc738d8, 0xc5fe49)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/proc.c:1385 +0x3b
runfinq()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/mgc0.c:2644 +0xcf
runtime.goexit()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/proc.c:1445
goroutine 338 [sleep]:
time.Sleep(0x989680)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/time.goc:39 +0x31
github.com/coreos/etcd/integration.(_cluster).waitMembersMatch(0xc20853d9c0, 0xc20804e000, 0xc208090000, 0x3, 0x3)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:327 +0x40a
github.com/coreos/etcd/integration.(_cluster).Launch(0xc20853d9c0, 0xc20804e000)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:217 +0x365
github.com/coreos/etcd/integration.testClusterUsingDiscovery(0xc20804e000, 0x3)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:86 +0x59a
github.com/coreos/etcd/integration.TestClusterOf3UsingDiscovery(0xc20804e000)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:69 +0x3e
testing.tRunner(0xc20804e000, 0xc5d828)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:422 +0x110
created by testing.RunTests
    /home/travis/.gvm/gos/go1.3.3/src/pkg/testing/testing.go:504 +0xb46
goroutine 429 [IO wait]:
net.runtime_pollWait(0x7f9909e4e6d0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6530, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6530, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b64d0, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e678, 0xc20803e490, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e678, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2085e9440, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc2080c7080, 0x7f9909e48290, 0xc2085e9440, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 428 [IO wait]:
net.runtime_pollWait(0x7f9909e4e200, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b63e0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b63e0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b6380, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e670, 0xc20803e940, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e670, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2085e9380, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc2080c6f60, 0x7f9909e48290, 0xc2085e9380, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 447 [select]:
net/http.(_persistConn).writeLoop(0xc208043ef0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:885 +0x483
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:601 +0xd5f
goroutine 411 [IO wait]:
net.runtime_pollWait(0x7f9909e475d0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6d10, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6d10, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b6cb0, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e698, 0x7f9909e36000, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e698, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2080ac840, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc208520ae0, 0x7f9909e48290, 0xc2080ac840, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 446 [IO wait]:
net.runtime_pollWait(0x7f9909e47520, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc20810c290, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc20810c290, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc20810c230, 0xc2080cc000, 0x1000, 0x1000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e480, 0xc2080cc000, 0x1000, 0x1000, 0xc75840, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e480, 0xc2080cc000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:23 +0x78
github.com/coreos/etcd/pkg/transport.timeoutConn.Read(0x7f9909e48648, 0xc20803e480, 0x12a05f200, 0x12a05f200, 0xc2080cc000, 0x1000, 0x1000, 0x7f9909e3e0f8, 0x0, 0x0)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/pkg/transport/timeout_conn.go:45 +0x18b
github.com/coreos/etcd/pkg/transport.(_timeoutConn).Read(0xc20800e500, 0xc2080cc000, 0x1000, 0x1000, 0x426038, 0x0, 0x0)
    <autogenerated>:13 +0xf4
net/http.noteEOFReader.Read(0x7f9909e487a8, 0xc20800e500, 0xc208043f48, 0xc2080cc000, 0x1000, 0x1000, 0xc2080a8ab0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:1203 +0x80
net/http.(_noteEOFReader).Read(0xc20800e520, 0xc2080cc000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:124 +0xf2
bufio.(_Reader).fill(0xc2080056e0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).Peek(0xc2080056e0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:132 +0x1ac
net/http.(_persistConn).readLoop(0xc208043ef0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:782 +0xf9
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:600 +0xd44
goroutine 448 [IO wait]:
net.runtime_pollWait(0x7f9909e46a20, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc20810c7d0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc20810c7d0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc20810c770, 0xc2080aa000, 0x1000, 0x1000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e490, 0xc2080aa000, 0x1000, 0x1000, 0x4c94ce, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e490, 0xc2080aa000, 0x1000, 0x1000, 0xc2080535c8, 0x0, 0x0)
    <autogenerated>:23 +0x78
net/http.(_liveSwitchReader).Read(0xc208051ca8, 0xc2080aa000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:206 +0xd3
io.(_LimitedReader).Read(0xc20800e660, 0xc2080aa000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:399 +0x17e
bufio.(_Reader).fill(0xc208005740)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).ReadSlice(0xc208005740, 0xc4420a, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:298 +0x5bf
bufio.(_Reader).ReadLine(0xc208005740, 0x0, 0x0, 0x0, 0xc20810c700, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:326 +0x7a
net/textproto.(_Reader).readLineSlice(0xc2080a8a20, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:55 +0xc2
net/textproto.(_Reader).ReadLine(0xc2080a8a20, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:36 +0x5c
net/http.ReadRequest(0xc208005740, 0xc2081195f0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/request.go:556 +0xd8
net/http.(_conn).readRequest(0xc208051c80, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:577 +0x3e6
net/http.(_conn).serve(0xc208051c80)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1132 +0x8d3
created by net/http.(_Server).Serve
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1721 +0x35d
goroutine 443 [IO wait]:
net.runtime_pollWait(0x7f9909e4e620, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6920, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6920, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b68c0, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e688, 0xc7faf8, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e688, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc208531560, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc2080048a0, 0x7f9909e48290, 0xc208531560, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 442 [IO wait]:
net.runtime_pollWait(0x7f9909e46ce0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b67d0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b67d0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b6770, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e680, 0xc20803e5a0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e680, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2085314a0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc2080047e0, 0x7f9909e48290, 0xc2085314a0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 380 [select]:
net/http.(_persistConn).writeLoop(0xc2080424d0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:885 +0x483
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:601 +0xd5f
goroutine 343 [select]:
github.com/coreos/etcd/etcdserver.(_EtcdServer).purgeFile(0xc2080903c0)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:359 +0x559
created by github.com/coreos/etcd/etcdserver.(_EtcdServer).Start
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:331 +0x72
goroutine 401 [IO wait]:
net.runtime_pollWait(0x7f9909e4e360, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b7250, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b7250, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b71f0, 0xc2085dc000, 0x8000, 0x8000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e110, 0xc2085dc000, 0x8000, 0x8000, 0x78, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e110, 0xc2085dc000, 0x8000, 0x8000, 0x78, 0x0, 0x0)
    <autogenerated>:23 +0x78
io.Copy(0x7f9909e491e0, 0xc20803e728, 0x7f9909e486a8, 0xc20803e110, 0xfd, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:353 +0x22e
net/http.func·008()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:163 +0xb8
created by net/http.(_conn).closeNotify
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:169 +0x3ab
goroutine 393 [IO wait]:
net.runtime_pollWait(0x7f9909e471b0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b7330, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b7330, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b72d0, 0xc208516000, 0x8000, 0x8000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e128, 0xc208516000, 0x8000, 0x8000, 0x8d, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e128, 0xc208516000, 0x8000, 0x8000, 0x8d, 0x0, 0x0)
    <autogenerated>:23 +0x78
io.Copy(0x7f9909e491e0, 0xc20803e408, 0x7f9909e486a8, 0xc20803e128, 0x217, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:353 +0x22e
net/http.func·008()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:163 +0xb8
created by net/http.(_conn).closeNotify
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:169 +0x3ab
goroutine 381 [IO wait]:
net.runtime_pollWait(0x7f9909e47890, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6760, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6760, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b6700, 0xc20809e000, 0x1000, 0x1000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e108, 0xc20809e000, 0x1000, 0x1000, 0xc2080e254a, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e108, 0xc20809e000, 0x1000, 0x1000, 0xc20804f320, 0x0, 0x0)
    <autogenerated>:23 +0x78
net/http.noteEOFReader.Read(0x7f9909e486a8, 0xc20803e108, 0xc2080425d8, 0xc20809e000, 0x1000, 0x1000, 0xc2085e8bd0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:1203 +0x80
net/http.(_noteEOFReader).Read(0xc2080edc40, 0xc20809e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:124 +0xf2
bufio.(_Reader).fill(0xc2080c7e00)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).Peek(0xc2080c7e00, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:132 +0x1ac
net/http.(_persistConn).readLoop(0xc208042580)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:782 +0xf9
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:600 +0xd44
goroutine 410 [IO wait]:
net.runtime_pollWait(0x7f9909e4e780, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6bc0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6bc0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc2080b6b60, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e690, 0xc20803e568, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e690, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2080ac780, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc208520a80, 0x7f9909e48290, 0xc2080ac780, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 378 [select]:
net/http.(_persistConn).writeLoop(0xc208042000)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:885 +0x483
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:601 +0xd5f
goroutine 345 [IO wait]:
net.runtime_pollWait(0x7f9909e4e4c0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2085221b0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2085221b0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc208522150, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e060, 0xc20803e128, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e060, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2080accf0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc2080053e0, 0x7f9909e48290, 0xc2080accf0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 385 [semacquire]:
sync.runtime_Syncsemacquire(0xc208058340)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/sema.goc:257 +0xc0
sync.(_Cond).Wait(0xc208058330)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/sync/cond.go:62 +0xb5
io.(_pipe).read(0xc208058300, 0xc208560000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:52 +0x40c
io.(_PipeReader).Read(0xc20803e400, 0xc208560000, 0x1000, 0x1000, 0xc44200, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:134 +0x7e
net/http.(_liveSwitchReader).Read(0xc208050628, 0xc208560000, 0x1000, 0x1000, 0x4, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:206 +0xd3
io.(_LimitedReader).Read(0xc2080ede00, 0xc208560000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:399 +0x17e
bufio.(_Reader).fill(0xc2080c60c0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).ReadSlice(0xc2080c60c0, 0xc4420a, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:298 +0x5bf
bufio.(_Reader).ReadLine(0xc2080c60c0, 0x0, 0x0, 0x0, 0xc2080b7200, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:326 +0x7a
net/textproto.(_Reader).readLineSlice(0xc2080ad320, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:55 +0xc2
net/textproto.(_Reader).ReadLine(0xc2080ad320, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:36 +0x5c
net/http.ReadRequest(0xc2080c60c0, 0xc20810b2b0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/request.go:556 +0xd8
net/http.(_conn).readRequest(0xc208050600, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:577 +0x3e6
net/http.(_conn).serve(0xc208050600)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1132 +0x8d3
created by net/http.(_Server).Serve
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1721 +0x35d
goroutine 400 [IO wait]:
net.runtime_pollWait(0x7f9909e4e2b0, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b72c0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b72c0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b7260, 0xc208550000, 0x8000, 0x8000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e120, 0xc208550000, 0x8000, 0x8000, 0x78, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e120, 0xc208550000, 0x8000, 0x8000, 0x78, 0x0, 0x0)
    <autogenerated>:23 +0x78
io.Copy(0x7f9909e491e0, 0xc20803e630, 0x7f9909e486a8, 0xc20803e120, 0xfd, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:353 +0x22e
net/http.func·008()
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:163 +0xb8
created by net/http.(_conn).closeNotify
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:169 +0x3ab
goroutine 377 [IO wait]:
net.runtime_pollWait(0x7f9909e4e990, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b6a00, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b6a00, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b69a0, 0xc20855d000, 0x1000, 0x1000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e0f0, 0xc20855d000, 0x1000, 0x1000, 0xc2080bc8a9, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e0f0, 0xc20855d000, 0x1000, 0x1000, 0xc20804f3b0, 0x0, 0x0)
    <autogenerated>:23 +0x78
net/http.noteEOFReader.Read(0x7f9909e486a8, 0xc20803e0f0, 0xc208042058, 0xc20855d000, 0x1000, 0x1000, 0xc2080ad7a0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:1203 +0x80
net/http.(_noteEOFReader).Read(0xc2080ed8a0, 0xc20855d000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:124 +0xf2
bufio.(_Reader).fill(0xc2080c7020)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).Peek(0xc2080c7020, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:132 +0x1ac
net/http.(_persistConn).readLoop(0xc208042000)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:782 +0xf9
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:600 +0xd44
goroutine 382 [select]:
net/http.(_persistConn).writeLoop(0xc208042580)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:885 +0x483
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:601 +0xd5f
goroutine 340 [select]:
github.com/coreos/etcd/raft.(_node).run(0xc2080d6370, 0xc208042210)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:262 +0x1078
created by github.com/coreos/etcd/raft.StartNode
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/raft/node.go:168 +0x74d
goroutine 383 [semacquire]:
sync.runtime_Syncsemacquire(0xc208058c40)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/sema.goc:257 +0xc0
sync.(_Cond).Wait(0xc208058c30)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/sync/cond.go:62 +0xb5
io.(_pipe).read(0xc208058c00, 0xc208063000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:52 +0x40c
io.(_PipeReader).Read(0xc20803e6f0, 0xc208063000, 0x1000, 0x1000, 0xc208052768, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:134 +0x7e
net/http.(_liveSwitchReader).Read(0xc2080503a8, 0xc208063000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:206 +0xd3
io.(_LimitedReader).Read(0xc2080edcc0, 0xc208063000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:399 +0x17e
bufio.(_Reader).fill(0xc2080c7e60)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).ReadSlice(0xc2080c7e60, 0xc4420a, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:298 +0x5bf
bufio.(_Reader).ReadLine(0xc2080c7e60, 0x0, 0x0, 0x0, 0xc2080b7100, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:326 +0x7a
net/textproto.(_Reader).readLineSlice(0xc2085e8b40, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:55 +0xc2
net/textproto.(_Reader).ReadLine(0xc2085e8b40, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:36 +0x5c
net/http.ReadRequest(0xc2080c7e60, 0xc20810bee0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/request.go:556 +0xd8
net/http.(_conn).readRequest(0xc208050380, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:577 +0x3e6
net/http.(_conn).serve(0xc208050380)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1132 +0x8d3
created by net/http.(_Server).Serve
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1721 +0x35d
goroutine 344 [IO wait]:
net.runtime_pollWait(0x7f9909e47310, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc208522060, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc208522060, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).accept(0xc208522000, 0xa93d90, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:419 +0x435
net.(_TCPListener).AcceptTCP(0xc20803e058, 0xc2085344b0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:234 +0x8b
net.(_TCPListener).Accept(0xc20803e058, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/tcpsock_posix.go:244 +0x59
net/http/httptest.(_historyListener).Accept(0xc2080ac9f0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:48 +0xa0
net/http.(_Server).Serve(0xc208005380, 0x7f9909e48290, 0xc2080ac9f0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1698 +0xa2
created by net/http/httptest.(_Server).Start
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/httptest/server.go:109 +0x44e
goroutine 341 [select]:
github.com/coreos/etcd/etcdserver.(_EtcdServer).run(0xc2080903c0)
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:408 +0x1215
created by github.com/coreos/etcd/etcdserver.(_EtcdServer).start
    /home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server.go:348 +0x2b4
goroutine 384 [semacquire]:
sync.runtime_Syncsemacquire(0xc2080581c0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/sema.goc:257 +0xc0
sync.(_Cond).Wait(0xc2080581b0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/sync/cond.go:62 +0xb5
io.(_pipe).read(0xc208058180, 0xc2080de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:52 +0x40c
io.(_PipeReader).Read(0xc20803e628, 0xc2080de000, 0x1000, 0x1000, 0xc208052628, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/pipe.go:134 +0x7e
net/http.(_liveSwitchReader).Read(0xc208050428, 0xc2080de000, 0x1000, 0x1000, 0x1, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:206 +0xd3
io.(_LimitedReader).Read(0xc2080edd20, 0xc2080de000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/io/io.go:399 +0x17e
bufio.(_Reader).fill(0xc2080c7f20)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).ReadSlice(0xc2080c7f20, 0xc4420a, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:298 +0x5bf
bufio.(_Reader).ReadLine(0xc2080c7f20, 0x0, 0x0, 0x0, 0xc2080b7200, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:326 +0x7a
net/textproto.(_Reader).readLineSlice(0xc2085e8960, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:55 +0xc2
net/textproto.(_Reader).ReadLine(0xc2085e8960, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/textproto/reader.go:36 +0x5c
net/http.ReadRequest(0xc2080c7f20, 0xc20810b930, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/request.go:556 +0xd8
net/http.(_conn).readRequest(0xc208050400, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:577 +0x3e6
net/http.(_conn).serve(0xc208050400)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1132 +0x8d3
created by net/http.(_Server).Serve
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/server.go:1721 +0x35d
goroutine 379 [IO wait]:
net.runtime_pollWait(0x7f9909e47940, 0x72, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/runtime/netpoll.goc:146 +0x66
net.(_pollDesc).Wait(0xc2080b68b0, 0x72, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:84 +0x62
net.(_pollDesc).WaitRead(0xc2080b68b0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_poll_runtime.go:89 +0x50
net.(_netFD).Read(0xc2080b6850, 0xc208564000, 0x1000, 0x1000, 0x0, 0x7f9909e463f0, 0xb)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/fd_unix.go:242 +0x430
net.(_conn).Read(0xc20803e100, 0xc208564000, 0x1000, 0x1000, 0xc2080e3179, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/net.go:122 +0x12d
net.(_TCPConn).Read(0xc20803e100, 0xc208564000, 0x1000, 0x1000, 0xc20804fef0, 0x0, 0x0)
    <autogenerated>:23 +0x78
net/http.noteEOFReader.Read(0x7f9909e486a8, 0xc20803e100, 0xc208042528, 0xc208564000, 0x1000, 0x1000, 0xc208530e40, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:1203 +0x80
net/http.(_noteEOFReader).Read(0xc2080ed9e0, 0xc208564000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
    <autogenerated>:124 +0xf2
bufio.(_Reader).fill(0xc2080c7d40)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:97 +0x35d
bufio.(_Reader).Peek(0xc2080c7d40, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/bufio/bufio.go:132 +0x1ac
net/http.(_persistConn).readLoop(0xc2080424d0)
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:782 +0xf9
created by net/http.(_Transport).dialConn
    /home/travis/.gvm/gos/go1.3.3/src/pkg/net/http/transport.go:600 +0xd44

--- a/discovery/discovery.go
+++ b/discovery/discovery.go
@@ -276,7 +276,8 @@ func (d *discovery) waitNodes(nodes client.Nodes, size int, index uint64) (clien
 	if len(nodes) > size {
 		nodes = nodes[:size]
 	}
-	w := d.c.RecursiveWatch(d.cluster, index)
+	// watch from the next index
+	w := d.c.RecursiveWatch(d.cluster, index+1)
 	all := make(client.Nodes, len(nodes))
 	copy(all, nodes)
 	for _, n := range all {

